src/functions/onDisconnect.ts

```diff
@@ -4,6 +4,35 @@ import { response } from '../utils/responseHelper';
 import { Logger } from '../utils/Logger';
 import { WebSocketAPIGatewayEvent } from '../types/event';
 import { LambdaResponse } from '../types/response';
+import { deleteGame, removeUserFromGame } from '../module/gameDBService';
+import { User } from '../models/User';
+
+/* ----------------------------------------------------------------------------
+ * Handler Helper Functions
+ * ------------------------------------------------------------------------- */
+
+/**
+ * Additional cleanup when user disconnect
+ * @param deletedUser
+ */
+const cleanupUserWhenDisconnect = async (deletedUser: User) => {
+  // TODO: Optimize this cleanup process
+  const { gameId, connectionId } = deletedUser;
+
+  // 1. Remove user from game (if there is gameId attribute on the user)
+  if (gameId) {
+    const updatedGame = await removeUserFromGame(gameId, connectionId);
+
+    // 2. Delete the game if the user is the host of the game
+    if (updatedGame && updatedGame.host.connectionId === connectionId) {
+      await deleteGame(gameId);
+    }
+  }
+};
```

Comment:
Probably should follow the same logic as leaveGame. Need to send updates to other connections if they are in the same game.
Also, need to do a check on the game.started value. If true, we do not want to delete the game, and disrupt the game while others are in it.

Reply:
`Probably should follow the same logic as leaveGame. Need to send updates to other connections if they are in the same game.`: sure
`Also, need to do a check on the game.started value. If true, we do not want to delete the game, and disrupt the game while others are in it.`: do we want to delete the game when all user leaves or just switch host to another user randomly?

Reply:
Right now, our implementation is if the user is host, then the game gets deleted. Do you want to work on changing the host or keep it as is?

If changing,
If game started = false,
1. change host
2. send in_game_update { connections and host }
3. send in_game_message (someone left)

If game started = true,
1. don't know if changing host makes a difference but we can implement it for some use in the future
2. we do not have listeners in the front-end yet for when Game has started (can leave this empty)

Reply:
^ Above applies if user is host
If not changing,
If game started = false,
1. delete game
2. send LEAVE_GAME to current connections who are in game

If game started = true,
1. don't delete game

Reply:
> ^ Above applies if user is host
> If not changing,
> If game started = false,
> 
> 1. delete game
> 2. send LEAVE_GAME to current connections who are in game
> 
> If game started = true,
> 
> 1. don't delete game

I think I will keep it as it is for now :)
Regarding this
```
If game started = true,

don't delete game
```
so I don't delete the game, maybe set the game `started` attribute back to false, and wait for another user to join in order to start a new game? or do the remaining users have to leave the game and join a new one (because there is no host in the game if we use the original approach)?

Reply:
Depends. We can think of how we want to handle this when we are developing the game.
we can either:
1. make an AI to take over the player
2. make the game available so others can take over the current player
3. stop the game completing and kick everybody out

Reply:
So on a logical level, if the game hasn't started, the flow of checking if host or not host should be the same as the current implementation of onLeaveGame

Reply:
> Depends. We can think of how we want to handle this when we are developing the game.
> we can either:
> 
> 1. make an AI to take over the player
> 2. make the game available so others can take over the current player
> 3. stop the game completing and kick everybody out

I can do option 3 for now, but option 2 is viable too

Reply:
> So on a logical level, if the game hasn't started, the flow of checking if host or not host should be the same as the current implementation of onLeaveGame

yup, I will use the same helper function to send the updates to user (both onLeaveGame and onDisconnect)

Reply:
Works well on FE